src/common/warnings.ts

```diff
@@ -0,0 +1,41 @@
+export enum SDSWarningTypes {
+  ButtonMissingSDSProps = "buttonMissingProps",
+  ChipDeprecated = "chipDeprecated",
+  MenuSelectDeprecated = "menuSelectDeprecated",
+}
+
+const SDS_WARNINGS = {
+  [SDSWarningTypes.ButtonMissingSDSProps]: {
+    hasWarned: false,
+    message:
+      "Warning: Buttons without sdsStyle or sdsType props will be deprecated.",
+  },
+  [SDSWarningTypes.ChipDeprecated]: {
+    hasWarned: false,
+    message: "Warning: <Chip /> will be deprecated and replaced with <Tag />",
+  },
+  [SDSWarningTypes.MenuSelectDeprecated]: {
+    hasWarned: false,
+    message:
+      "Warning: MenuSelect will be deprecated and replaced with <DropdownMenu />",
+  },
+};
+
+export const showWarningIfFirstOccurence = (warningType: SDSWarningTypes) => {
+  if (!(<string>warningType in SDS_WARNINGS)) {
```

Comment:
FYI, if you actually add a new type in the `SDSWarningTypes` enum without adding it to the `SDS_WARNINGS` object, there are a number of typescript errors that are raised in this method, but this warning will make it explicit in the browser as well.